Port to ROS 2

vmayoral opened the pull request:
**Note to maintainers**: Please do not merge in this branch, create a ros2 branch and modify this PR.

With the changes provided, package compiles in a ROS 2 workspace. 

Pending items:
- [x] Review warnings
- [ ] Port testing

For completeness, warnings pending of review:
```bash
In file included from /Users/victor/ros2_moveit_ws/src/geometric_shapes/src/shapes.cpp:39:
In file included from /usr/local/include/octomap/octomap.h:37:
In file included from /usr/local/include/octomap/OcTree.h:38:
In file included from /usr/local/include/octomap/OccupancyOcTreeBase.h:506:
/usr/local/include/octomap/OccupancyOcTreeBase.hxx:116:108: warning: unused parameter 'maxrange' [-Wunused-parameter]
  void OccupancyOcTreeBase<NODE>::insertPointCloudRays(const Pointcloud& pc, const point3d& origin, double maxrange, bool lazy_eval) {
                                                                                                           ^
In file included from /Users/victor/ros2_moveit_ws/src/geometric_shapes/src/shapes.cpp:40:
/usr/local/include/console_bridge/console.h:70:88: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
  console_bridge::log(__FILE__, __LINE__, console_bridge::CONSOLE_BRIDGE_LOG_WARN,  fmt, ##__VA_ARGS__)
                                                                                       ^
/Users/victor/ros2_moveit_ws/src/geometric_shapes/src/shapes.cpp:265:34: warning: unused parameter 'scale' [-Wunused-parameter]
void OcTree::scaleAndPadd(double scale, double padd)
                                 ^
/Users/victor/ros2_moveit_ws/src/geometric_shapes/src/shapes.cpp:265:48: warning: unused parameter 'padd' [-Wunused-parameter]
void OcTree::scaleAndPadd(double scale, double padd)
                                               ^
In file included from /Users/victor/ros2_moveit_ws/src/geometric_shapes/src/shapes.cpp:40:
/usr/local/include/console_bridge/console.h:70:88: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
  console_bridge::log(__FILE__, __LINE__, console_bridge::CONSOLE_BRIDGE_LOG_WARN,  fmt, ##__VA_ARGS__)
                                                                                       ^
/Users/victor/ros2_moveit_ws/src/geometric_shapes/src/shapes.cpp:270:33: warning: unused parameter 'scale' [-Wunused-parameter]
void Plane::scaleAndPadd(double scale, double padding)
                                ^
/Users/victor/ros2_moveit_ws/src/geometric_shapes/src/shapes.cpp:270:47: warning: unused parameter 'padding' [-Wunused-parameter]
void Plane::scaleAndPadd(double scale, double padding)
                                              ^
In file included from /Users/victor/ros2_moveit_ws/src/geometric_shapes/src/mesh_operations.cpp:49:
/usr/local/include/assimp/scene.h:240:1: warning: 'aiScene' defined as a struct here but previously declared as a class [-Wmismatched-tags]
struct aiScene
^
/Users/victor/ros2_moveit_ws/src/geometric_shapes/include/geometric_shapes/mesh_operations.h:45:1: note: did you mean struct here?
class aiScene;
^~~~~
struct
In file included from /Users/victor/ros2_moveit_ws/src/geometric_shapes/src/body_operations.cpp:39:
/usr/local/include/console_bridge/console.h:67:90: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
  console_bridge::log(__FILE__, __LINE__, console_bridge::CONSOLE_BRIDGE_LOG_ERROR, fmt, ##__VA_ARGS__)
                                                                                         ^
In file included from /Users/victor/ros2_moveit_ws/src/geometric_shapes/src/shape_operations.cpp:45:
/usr/local/include/console_bridge/console.h:70:88: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
  console_bridge::log(__FILE__, __LINE__, console_bridge::CONSOLE_BRIDGE_LOG_WARN,  fmt, ##__VA_ARGS__)
                                                                                       ^
/usr/local/include/console_bridge/console.h:67:90: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
  console_bridge::log(__FILE__, __LINE__, console_bridge::CONSOLE_BRIDGE_LOG_ERROR, fmt, ##__VA_ARGS__)
                                                                                         ^
/usr/local/include/console_bridge/console.h:67:88: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
  console_bridge::log(__FILE__, __LINE__, console_bridge::CONSOLE_BRIDGE_LOG_ERROR, fmt, ##__VA_ARGS__)
                                                                                       ^
/Users/victor/ros2_moveit_ws/src/geometric_shapes/src/shape_operations.cpp:158:49: warning: unused parameter 'shape_msg' [-Wunused-parameter]
  void operator()(const shape_msgs::msg::Plane& shape_msg) const
                                                ^
In file included from /Users/victor/ros2_moveit_ws/src/geometric_shapes/src/shape_operations.cpp:45:
/usr/local/include/console_bridge/console.h:67:90: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
  console_bridge::log(__FILE__, __LINE__, console_bridge::CONSOLE_BRIDGE_LOG_ERROR, fmt, ##__VA_ARGS__)
                                                                                         ^
/Users/victor/ros2_moveit_ws/src/geometric_shapes/src/shape_operations.cpp:205:60: warning: unused parameter 'shape_msg' [-Wunused-parameter]
  Eigen::Vector3d operator()(const shape_msgs::msg::Plane& shape_msg) const
                                                           ^
In file included from /Users/victor/ros2_moveit_ws/src/geometric_shapes/src/shape_operations.cpp:45:
/usr/local/include/console_bridge/console.h:67:90: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
  console_bridge::log(__FILE__, __LINE__, console_bridge::CONSOLE_BRIDGE_LOG_ERROR, fmt, ##__VA_ARGS__)
                                                                                         ^
/usr/local/include/console_bridge/console.h:67:90: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
/usr/local/include/console_bridge/console.h:67:90: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
7 warnings generated.
/Users/victor/ros2_moveit_ws/src/geometric_shapes/src/bodies.cpp:131:67: warning: unused parameter 'verbose' [-Wunused-parameter]
bool bodies::Sphere::containsPoint(const Eigen::Vector3d& p, bool verbose) const
                                                                  ^
In file included from /Users/victor/ros2_moveit_ws/src/geometric_shapes/src/mesh_operations.cpp:46:
/usr/local/include/console_bridge/console.h:67:88: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
  console_bridge::log(__FILE__, __LINE__, console_bridge::CONSOLE_BRIDGE_LOG_ERROR, fmt, ##__VA_ARGS__)
                                                                                       ^
/usr/local/include/console_bridge/console.h:70:88: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
  console_bridge::log(__FILE__, __LINE__, console_bridge::CONSOLE_BRIDGE_LOG_WARN,  fmt, ##__VA_ARGS__)
                                                                                       ^
/usr/local/include/console_bridge/console.h:67:90: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
  console_bridge::log(__FILE__, __LINE__, console_bridge::CONSOLE_BRIDGE_LOG_ERROR, fmt, ##__VA_ARGS__)
                                                                                         ^
/usr/local/include/console_bridge/console.h:70:90: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
  console_bridge::log(__FILE__, __LINE__, console_bridge::CONSOLE_BRIDGE_LOG_WARN,  fmt, ##__VA_ARGS__)
                                                                                         ^
/usr/local/include/console_bridge/console.h:70:90: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
/usr/local/include/console_bridge/console.h:70:88: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
  console_bridge::log(__FILE__, __LINE__, console_bridge::CONSOLE_BRIDGE_LOG_WARN,  fmt, ##__VA_ARGS__)
                                                                                       ^
/usr/local/include/console_bridge/console.h:70:90: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
  console_bridge::log(__FILE__, __LINE__, console_bridge::CONSOLE_BRIDGE_LOG_WARN,  fmt, ##__VA_ARGS__)
                                                                                         ^
/usr/local/include/console_bridge/console.h:70:90: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
/usr/local/include/console_bridge/console.h:70:90: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
/usr/local/include/console_bridge/console.h:67:90: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
  console_bridge::log(__FILE__, __LINE__, console_bridge::CONSOLE_BRIDGE_LOG_ERROR, fmt, ##__VA_ARGS__)
                                                                                         ^
/Users/victor/ros2_moveit_ws/src/geometric_shapes/src/bodies.cpp:264:69: warning: unused parameter 'verbose' [-Wunused-parameter]
bool bodies::Cylinder::containsPoint(const Eigen::Vector3d& p, bool verbose) const
                                                                    ^
/Users/victor/ros2_moveit_ws/src/geometric_shapes/src/bodies.cpp:317:99: warning: unused parameter 'max_attempts' [-Wunused-parameter]
bool bodies::Cylinder::samplePointInside(random_numbers::RandomNumberGenerator& rng, unsigned int max_attempts,
                                                                                                  ^
/Users/victor/ros2_moveit_ws/src/geometric_shapes/src/bodies.cpp:477:64: warning: unused parameter 'verbose' [-Wunused-parameter]
bool bodies::Box::containsPoint(const Eigen::Vector3d& p, bool verbose) const
                                                               ^
/Users/victor/ros2_moveit_ws/src/geometric_shapes/src/bodies.cpp:665:71: warning: unused parameter 'verbose' [-Wunused-parameter]
bool bodies::ConvexMesh::containsPoint(const Eigen::Vector3d& p, bool verbose) const
                                                                      ^
In file included from /Users/victor/ros2_moveit_ws/src/geometric_shapes/src/bodies.cpp:40:
/usr/local/include/console_bridge/console.h:70:88: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
  console_bridge::log(__FILE__, __LINE__, console_bridge::CONSOLE_BRIDGE_LOG_WARN,  fmt, ##__VA_ARGS__)
                                                                                       ^
/Users/victor/ros2_moveit_ws/src/geometric_shapes/src/mesh_operations.cpp:633:13: warning: unused function 'writeFloatToSTL' [-Wunused-function]
inline void writeFloatToSTL(char*& ptr, float data)
            ^
/usr/local/include/console_bridge/console.h:67:90: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
  console_bridge::log(__FILE__, __LINE__, console_bridge::CONSOLE_BRIDGE_LOG_ERROR, fmt, ##__VA_ARGS__)
                                                                                         ^
/usr/local/include/console_bridge/console.h:67:90: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
2 warnings generated.
8 warnings generated.
12 warnings generated.
8 warnings generated.
```

vmayoral committed:
Port headers to ROS 2

vmayoral committed:
Adapt shape_msgs::SolidPrimitive to ROS 2

Replaced by shape_msgs::msg::SolidPrimitive

vmayoral committed:
Adapt shape_msgs::Mesh to ROS 2

vmayoral committed:
Adapt visualization_msgs::Marker to ROS 2

vmayoral committed:
Adapt geometry_msgs usage to ROS 2

vmayoral committed:
Adapt a variety of primitives to ROS 2

vmayoral committed:
Port CMakeLists.txt and package.xml

rhaschke commented:
Directed at ros2 branch.

anasarrak committed:
Update setpose to be used with Affine3d

vmayoral committed:
Few bits for porting build testing to ROS 2

Requires further work

vmayoral committed:
Merge branch 'ros2' of https://github.com/AcutronicRobotics/geometric_shapes into ros2

vmayoral commented:
Ported a few bits of the build testing infrastructure but it requires further work. Help is welcome @rhaschke, @davetcoleman, @gavanderhoorn and others.

anasarrak committed:
Change bodies Eigen::Issometry calls to Eigen::Affine3d

ahcorde committed:
Merge pull request #1 from anasarrak/ros2

Change bodies Eigen::Issometry calls to Eigen::Affine3d

gavanderhoorn commented on include/geometric_shapes/bodies.h:
```diff
@@ -499,21 +499,21 @@ class BodyVector
 
   /** \brief Construct a body vector from a vector of shapes, a vector of poses and a padding */
-  BodyVector(const std::vector<shapes::Shape*>& shapes, const EigenSTL::vector_Isometry3d& poses, double padding = 0.0);
+  BodyVector(const std::vector<shapes::Shape*>& shapes, const EigenSTL::vector_Affine3d& poses, double padding = 0.0);
```
I understand that this is currently blocking you in your porting efforts, but it seems more efficient and forward thinking to me to assume that the `Isometry3d` migration will happen in all ROS pkgs that use `Eigen::Affine3d`.

Instead of reverting all those changes in MoveIt and related pkgs, perhaps forward-porting the `Isometry3d` changes to the pkgs that *should* have it would be more efficient in the long run.

davetcoleman replied:
+1 we do not want to revert back to affine3d

davetcoleman commented on CMakeLists.txt:
```diff
@@ -66,18 +57,25 @@ add_library(${PROJECT_NAME}
+#   find_package(ament_cmake_gtest REQUIRED)
+#   # Unit tests
+#   add_subdirectory(test)
+# endif()
```
enable tests

mlautman replied:
+1

davetcoleman commented on package.xml:
```diff
@@ -1,18 +1,22 @@
+<package format="3">
   <name>geometric_shapes</name>
-  <version>0.6.1</version>
+  <version>0.6.2</version>
```
is it necessary to bump the version here? my understanding is that only bloom does this during the release process @130s ?

130s replied:
It's ok to modify version number along with other code change although that sounds unusual.  IMO it's typically done during release flow, e.g. in Catkin's world, `catkin_prepare_release` nicely takes care of it automatically. Using a toolchain like that is preferable as it allows release flow to be more systematic and organized.

And the change suggested in this PR seems non-trivial, so bumping minor or even major number makes more sense to me (if the repo follows common major-minor-patch version semantics).

mlautman replied:
We should keep ROS and ROS2 versions separate. In the RQT port we changed all of the ROS2 versions to 1.0.0 although I am not sure that is ideal

anasarrak committed:
Return back to Isometry3d

ahcorde committed:
Merge pull request #2 from anasarrak/ros2

Return back to Isometry3d

mlautman commented on CMakeLists.txt:
```diff
@@ -66,18 +57,25 @@ add_library(${PROJECT_NAME}
-        ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-        LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION})
+        ARCHIVE DESTINATION lib
+      LIBRARY DESTINATION lib)
```
fix alignment

mlautman commented on CMakeLists.txt:
```diff
@@ -66,18 +57,25 @@ add_library(${PROJECT_NAME}
+        ARCHIVE DESTINATION lib
+      LIBRARY DESTINATION lib)
 
 install(DIRECTORY include/${PROJECT_NAME}/
```
you can use `include_directories(include)` no?

mlautman commented on include/geometric_shapes/bodies.h:
```diff
@@ -188,7 +188,7 @@ class Body
 
   /** \brief Get a clone of this body, but one that is located at the pose \e pose */
-  BodyPtr cloneAt(const Eigen::Isometry3d& pose) const
+  BodyPtr cloneAt(const Eigen::Isometry3d& pose)
```
Why are we mixing code changes with porting changes? Is this actually necessary for ROS2?

mlautman commented on include/geometric_shapes/bodies.h:
```diff
@@ -197,7 +197,7 @@ class Body
       padding and \e scaling. This function is useful to implement
       thread safety, when bodies need to be moved around. */
-  virtual BodyPtr cloneAt(const Eigen::Isometry3d& pose, double padding, double scaling) const = 0;
+  virtual BodyPtr cloneAt(const Eigen::Isometry3d& pose, double padding, double scaling);
```
Same as above.

mlautman commented on include/geometric_shapes/shape_extents.h:
```diff
@@ -35,16 +35,16 @@
 
-#include <shape_msgs/SolidPrimitive.h>
-#include <shape_msgs/Mesh.h>
+#include "shape_msgs/msg/solid_primitive.hpp"
```
Is there a reason to switch from <> to ""?

seanyen committed:
[Windows][melodic] Fix ASSIMP libraries path for Windows build (#101)

* consolidate finding absolute paths for ASSIMP_LIBRARIES

* Add comments.

Co-Authored-By: seanyen <seanyen@microsoft.com>

ahcorde committed:
Fix cmakelists to compile against other executables and fixing tests

ahcorde committed:
Converting into a SHARED library

vmayoral committed:
Merge pull request #3 from AcutronicRobotics/fix-cmake

Fix cmakelists to compile against other executables and fixing tests

vmayoral committed:
Add ASSIMP and console_bridge

reformat it slightly as well. This commit allows linking
in OS X but requires console_bridge to be installed from
sources.

Note that console_bridge_vendor is a shim over console_bridge
and still requires the installation of the later through its
sources.

kazuki0824 committed:
Fix packages.xml to fit with CMakeLists.txt.

LanderU committed:
Merge pull request #4 from kazuki0824/ros2

Fix packages.xml

peci1 committed:
Added bodies::Body::computeBoundingBox (aligned box version). (#104)

anasarrak committed:
Merge branch 'melodic-devel' into ros2

rhaschke commented:
Closing in favor of #118 / #122.

vmayoral commented:
@rhaschke tons of changes at #122 that removed authorship. E.g. @ahcorde contributed with https://github.com/ros-planning/geometric_shapes/pull/96/commits/25255f3fef31737c4881fcae91e511722411ad2d. This is wrong.
